src/hooks/use-settings-data.ts

```
@@ -0,0 +1,49 @@
+'use client';
+
+import { useState, useEffect } from 'react';
+import { useSettingsTab } from './use-settings-tab';
+
+interface UseSettingsDataOptions {
+    section: string;
+    defaultTab: string;
+    dataLoaders: {
+        [key: string]: () => Promise<any>;
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)